File: wlr/OGUR/OGUR/HUD/SkillHud.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using OGUR.Creatures;
using SPX.Util;
using SPX.Core;
using OGUR.Management;
using System;

namespace OGUR.HUD
{
    [Serializable]
    public class SkillHud:IHud
    {
        private static string __separator = "|";
        
        private Vector2 _manaPosition = new Vector2();        

        public SkillHud(ICreature owner) : base(owner,GameManager.SpriteWidth,XnaManager.WindowHeight / 4) 
        {
            _manaPosition = new Vector2(GetHudOrigin().X, GetHudOrigin().Y + XnaManager.WindowHeight / 4);
        }      

        private Vector2 CalculateHeight(string statType)
        {
            return new Vector2(_dimensions.X,(_parent.Get(statType) / _parent.GetMax(statType)) * _dimensions.Y);
        }

        private Vector2 CostOfCurrentSkill()
        {
            return new Vector2(_dimensions.X * .5f, _parent.GetCurrentSkillCost() / _parent.GetMax(StatType.MANA) * _dimensions.Y);
        }

        private string GetSkillStrings()
        {
            return "A:" + _parent.GetActiveSkillName() + __separator +
                     "X:" + _parent.GetHotSkillName(Commands.HotSkill1) + __separator +
                     "Y:" + _parent.GetHotSkillName(Commands.HotSkill2) + __separator +
                     "B:" + _parent.GetHotSkillName(Commands.HotSkill3) + __separator;
        }

        public void Update()
        {
            if (_isVisible)
            {
                _textHandler.Update();
                _textHandler.Clear();
                _textHandler.WriteDefault(GetSkillStrings(),GameManager.SpriteWidth, 0, GetHudOrigin());
            }
        }

        public void Draw()
        {
            if (!_isVisible) return;

            XnaManager.Renderer.Draw(_menuBase, GetHudOrigin(), new Rectangle(0, 0, 1, 1), Color.Green, 0f, Vector2.Zero, CalculateHeight(StatType.HEALTH), SpriteEffects.None, ZDepth.HudBG);
            XnaManager.Renderer.Draw(_menuBase, _manaPosition, new Rectangle(0, 0, 1, 1), Color.Yellow, 0f, Vector2.Zero, CostOfCurrentSkill(), SpriteEffects.None, ZDepth.HudBG);
            XnaManager.Renderer.Draw(_menuBase, _manaPosition, new Rectangle(0, 0, 1, 1), Color.Blue, 0f, Vector2.Zero, CalculateHeight(StatType.MANA), SpriteEffects.None, ZDepth.HudBG);

            _textHandler.Draw();
        }
    }
}

File: proj/src/aigilas/hud/SkillHud.java
package aigilas.hud;

import aigilas.creatures.BaseCreature;
import aigilas.creatures.StatType;
import aigilas.management.Commands;
import com.badlogic.gdx.graphics.Color;
import spx.bridge.DrawDepth;
import spx.core.Point2;
import spx.core.Settings;
import spx.core.SpxManager;

public class SkillHud extends BaseHud {
    private static final String __separator = "|";

    private Point2 _manaPosition = new Point2();

    public SkillHud(BaseCreature owner) {
        super(owner, Settings.get().spriteWidth, SpxManager.WindowHeight / 4);
        _manaPosition = new Point2(getHudOrigin().X, getHudOrigin().Y - SpxManager.WindowHeight / 4);
    }

    private int calculateHeight(StatType statType) {
        return (int) ((_parent.get(statType) / _parent.getMax(statType)) * _dimensions.Y);
    }

    private int costOfCurrentSkill() {
        return (int) (_parent.getCurrentSkillCost() / _parent.getMax(StatType.MANA) * _dimensions.Y);
    }

    private String getSkillStrings() {
        return "A:" + _parent.getActiveSkillName() + __separator + "X:" + _parent.getHotSkillName(Commands.HotSkill1) + __separator + "Y:" + _parent.getHotSkillName(Commands.HotSkill2) + __separator + "B:" + _parent.getHotSkillName(Commands.HotSkill3) + __separator;
    }

    public void update() {
        if (_isVisible) {
            _textHandler.update();
            _textHandler.clear();
            _textHandler.writeDefault(getSkillStrings(), (int) getHudOrigin().X + Settings.get().spriteWidth, Settings.get().spriteHeight, getHudOrigin());
        }
    }

    public void draw() {
        if (!_isVisible) {
            return;
        }

        SpxManager.Renderer.draw(_menuBase, getHudOrigin(), DrawDepth.HudBG, Color.GREEN, Settings.get().spriteWidth, calculateHeight(StatType.HEALTH));
        SpxManager.Renderer.draw(_menuBase, _manaPosition, DrawDepth.HudBG, Color.BLUE, Settings.get().spriteWidth, calculateHeight(StatType.MANA));
        SpxManager.Renderer.draw(_menuBase, _manaPosition, DrawDepth.HudBG, Color.YELLOW, Settings.get().spriteWidth / 2, costOfCurrentSkill());

        _textHandler.draw();
    }
}
